File: leetcode013/main.go
package main

import "fmt"

func main() {
	res := countBits(10)
	fmt.Println(res)
}

func countBits(n int) []int {
	ans, offset := make([]int, n+1), 1

	for i := 1; i <= n; i++ {
		if offset*2 == i {
			offset = i
		}
		ans[i] = 1 + ans[i-offset]

	}
	return ans
}

File: leetcode013/main.py
from typing import List


def countBits ( n:int) -> List[int] :
    dp = [0] * (n+1)
    offset = 1

    for i in range(1, n+ 1):
        n= 2
        if offset * 2 == i  :
            offset = i
        dp[i] = 1 + dp[i-offset]

    return dp

if __name__ == "__main__":
    res = countBits(10)
    print(res)

